### DISCUSSION TITLE
同一段代码，运行成功与否毫无规律可言

### DISCUSSION THREAD

**Discussion by sulv9999** [PaddlePaddle/PaddleOCR#13031] [Q&A]
#### 问题描述 / Problem Description
同一段代码，运行成功与否毫无规律可言，第一次成功第二次可能就不能成功，

#### 运行环境 / Runtime Environment
- OS:windows
- Paddle:
- PaddleOCR:2.7.3
- cuda11.8 cuDNN对应cuda11.8

#### 复现代码 / Reproduction Code

import os
from paddleocr import PaddleOCR, draw_ocr
import matplotlib.pyplot as plt
from PIL import Image
import cv2
import time
import numpy as np

class OCRProcessor:
    def __init__(self, det_model_dir, rec_model_dir, cls_model_dir, use_gpu=True, gpu_mem=4096, det_db_unclip_ratio=3):
        # 检查并创建模型目录
        model_dirs = [det_model_dir, rec_model_dir, cls_model_dir]
        for model_dir in model_dirs:
            if not os.path.exists(model_dir):
                os.makedirs(model_dir)

        # 初始化PaddleOCR
        self.ocr = PaddleOCR(
            use_angle_cls=True,
            lang='ch',
            use_gpu=use_gpu,
            det_model_dir=det_model_dir,
            rec_model_dir=rec_model_dir,
            cls_model_dir=cls_model_dir,
            gpu_mem=gpu_mem,
            det_db_unclip_ratio=det_db_unclip_ratio,
        )

    def process_image(self, img):
        # 读取图片
        # img = cv2.imread(img_path)

        # if img is None:
        #     raise FileNotFoundError(f"Image not found at the path: {img_path}")

        # 执行OCR识别
        startTime = time.time()
        result = self.ocr.ocr(img, cls=True)
        process_time = (time.time() - startTime) * 1000

        # 打印结果及耗时
        print(f"Processing time: {process_time} ms")
        for line in result:
            for e in line:
                print(e[0])
                print(e[1])

        return result, process_time

# 使用示例
if __name__ == '__main__':
    ocr_processor = OCRProcessor(
        det_model_dir='../ch_PP-OCRv4_det_server_infer',
        rec_model_dir='../ch_PP-OCRv4_rec_server_infer',
        cls_model_dir='../ch_ppocr_mobile_v2.0_cls_infer'
    )

    img_path = r'C:\Users\ly\Desktop\resuouce\orc\01.png'
    img = cv2.imread(img_path)
    result, process_time = ocr_processor.process_image(img)

#### 完整报错 / Complete Error Message
已连接到 pydev 调试器(内部版本号 241.14494.241)[2024/06/11 23:58:15] ppocr DEBUG: Namespace(alpha=1.0, alphacolor=(255, 255, 255), benchmark=False, beta=1.0, binarize=False, cls_batch_num=6, cls_image_shape='3, 48, 192', cls_model_dir='../ch_ppocr_mobile_v2.0_cls_infer', cls_thresh=0.9, cpu_threads=10, crop_res_save_dir='./output', det=True, det_algorithm='DB', det_box_type='quad', det_db_box_thresh=0.6, det_db_score_mode='fast', det_db_thresh=0.3, det_db_unclip_ratio=3, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_east_score_thresh=0.8, det_limit_side_len=960, det_limit_type='max', det_model_dir='../ch_PP-OCRv4_det_server_infer', det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, det_pse_thresh=0, det_sast_nms_thresh=0.2, det_sast_score_thresh=0.5, draw_img_save_dir='./inference_results', drop_score=0.5, e2e_algorithm='PGNet', e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_limit_side_len=768, e2e_limit_type='max', e2e_model_dir=None, e2e_pgnet_mode='fast', e2e_pgnet_score_thresh=0.5, e2e_pgnet_valid_set='totaltext', enable_mkldnn=False, fourier_degree=5, gpu_id=0, gpu_mem=4096, help='==SUPPRESS==', image_dir=None, image_orientation=False, invert=False, ir_optim=True, kie_algorithm='LayoutXLM', label_list=['0', '180'], lang='ch', layout=True, layout_dict_path=None, layout_model_dir=None, layout_nms_threshold=0.5, layout_score_threshold=0.5, max_batch_size=10, max_text_length=25, merge_no_span_structure=True, min_subgraph_size=15, mode='structure', ocr=True, ocr_order_method=None, ocr_version='PP-OCRv4', output='./output', page_num=0, precision='fp32', process_id=0, re_model_dir=None, rec=True, rec_algorithm='SVTR_LCNet', rec_batch_num=6, rec_char_dict_path='D:\\work\\work_space\\py\\ai\\ocr\\.venv\\lib\\site-packages\\paddleocr\\ppocr\\utils\\ppocr_keys_v1.txt', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_model_dir='../ch_PP-OCRv4_rec_server_infer', recovery=False, save_crop_res=False, save_log_path='./log_output/', scales=[8, 16, 32], ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ser_model_dir=None, show_log=True, sr_batch_num=1, sr_image_shape='3, 32, 128', sr_model_dir=None, structure_version='PP-StructureV2', table=True, table_algorithm='TableAttn', table_char_dict_path=None, table_max_len=488, table_model_dir=None, total_process_num=1, type='ocr', use_angle_cls=True, use_dilation=False, use_gpu=True, use_mp=False, use_npu=False, use_onnx=False, use_pdf2docx_api=False, use_pdserving=False, use_space_char=True, use_tensorrt=False, use_visual_backbone=True, use_xpu=False, vis_font_path='./doc/fonts/simfang.ttf', warmup=False)
[2024/06/11 23:58:23] ppocr DEBUG: dt_boxes num : 5, elapsed : 1.3452932834625244
Traceback (most recent call last):
  File "D:/apps/PyCharm 2024.1/plugins/python/helpers/pydev/pydevd.py", line 1535, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "D:\apps\PyCharm 2024.1\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "D:\work\work_space\py\ai\ocr\com.ly.ocr\ORCService.py", line 62, in <module>
    result, process_time = ocr_processor.process_image(img)
  File "D:\work\work_space\py\ai\ocr\com.ly.ocr\ORCService.py", line 39, in process_image
    result = self.ocr.ocr(img, cls=True)
  File "D:\work\work_space\py\ai\ocr\.venv\lib\site-packages\paddleocr\paddleocr.py", line 668, in ocr
    dt_boxes, rec_res, _ = self.__call__(img, cls)
  File "D:\work\work_space\py\ai\ocr\.venv\lib\site-packages\paddleocr\tools\infer\predict_system.py", line 99, in __call__
    img_crop_list, angle_list, elapse = self.text_classifier(
  File "D:\work\work_space\py\ai\ocr\.venv\lib\site-packages\paddleocr\tools\infer\predict_cls.py", line 111, in __call__
    self.predictor.run()
OSError: (External) CUBLAS error(7). 
  [Hint: 'CUBLAS_STATUS_INVALID_VALUE'.  An unsupported value or parameter was passed to the function (a negative vector size, for example). To correct: ensure that all the parameters being passed have valid values. ] (at ..\paddle/phi/kernels/funcs/blas/blas_impl.cu.h:41)
  [operator < fc > error]
  
  
  我没有做任何处理，完全没有做任何处理，只是在PyCharm 2024.1点击了重启，有时成功，有时失败
  
  
  这是成功的打印
  
  已连接到 pydev 调试器(内部版本号 241.14494.241)[2024/06/12 00:21:01] ppocr DEBUG: Namespace(alpha=1.0, alphacolor=(255, 255, 255), benchmark=False, beta=1.0, binarize=False, cls_batch_num=6, cls_image_shape='3, 48, 192', cls_model_dir='../ch_ppocr_mobile_v2.0_cls_infer', cls_thresh=0.9, cpu_threads=10, crop_res_save_dir='./output', det=True, det_algorithm='DB', det_box_type='quad', det_db_box_thresh=0.6, det_db_score_mode='fast', det_db_thresh=0.3, det_db_unclip_ratio=3, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_east_score_thresh=0.8, det_limit_side_len=960, det_limit_type='max', det_model_dir='../ch_PP-OCRv4_det_server_infer', det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, det_pse_thresh=0, det_sast_nms_thresh=0.2, det_sast_score_thresh=0.5, draw_img_save_dir='./inference_results', drop_score=0.5, e2e_algorithm='PGNet', e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_limit_side_len=768, e2e_limit_type='max', e2e_model_dir=None, e2e_pgnet_mode='fast', e2e_pgnet_score_thresh=0.5, e2e_pgnet_valid_set='totaltext', enable_mkldnn=False, fourier_degree=5, gpu_id=0, gpu_mem=4096, help='==SUPPRESS==', image_dir=None, image_orientation=False, invert=False, ir_optim=True, kie_algorithm='LayoutXLM', label_list=['0', '180'], lang='ch', layout=True, layout_dict_path=None, layout_model_dir=None, layout_nms_threshold=0.5, layout_score_threshold=0.5, max_batch_size=10, max_text_length=25, merge_no_span_structure=True, min_subgraph_size=15, mode='structure', ocr=True, ocr_order_method=None, ocr_version='PP-OCRv4', output='./output', page_num=0, precision='fp32', process_id=0, re_model_dir=None, rec=True, rec_algorithm='SVTR_LCNet', rec_batch_num=6, rec_char_dict_path='D:\\work\\work_space\\py\\ai\\ocr\\.venv\\lib\\site-packages\\paddleocr\\ppocr\\utils\\ppocr_keys_v1.txt', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_model_dir='../ch_PP-OCRv4_rec_server_infer', recovery=False, save_crop_res=False, save_log_path='./log_output/', scales=[8, 16, 32], ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ser_model_dir=None, show_log=True, sr_batch_num=1, sr_image_shape='3, 32, 128', sr_model_dir=None, structure_version='PP-StructureV2', table=True, table_algorithm='TableAttn', table_char_dict_path=None, table_max_len=488, table_model_dir=None, total_process_num=1, type='ocr', use_angle_cls=True, use_dilation=False, use_gpu=True, use_mp=False, use_npu=False, use_onnx=False, use_pdf2docx_api=False, use_pdserving=False, use_space_char=True, use_tensorrt=False, use_visual_backbone=True, use_xpu=False, vis_font_path='./doc/fonts/simfang.ttf', warmup=False)
[2024/06/12 00:21:09] ppocr DEBUG: dt_boxes num : 5, elapsed : 1.2375855445861816
[2024/06/12 00:21:09] ppocr DEBUG: cls num  : 5, elapsed : 0.015996694564819336
[2024/06/12 00:21:09] ppocr DEBUG: rec_res num  : 5, elapsed : 0.07001161575317383
Processing time: 1324.5902061462402 ms
[[53.0, 59.0], [99.0, 59.0], [99.0, 92.0], [53.0, 92.0]]
('大明', 0.9993781447410583)
[[53.0, 97.0], [99.0, 97.0], [99.0, 130.0], [53.0, 130.0]]
('大清', 0.9996690154075623)
[[53.0, 136.0], [100.0, 136.0], [100.0, 168.0], [53.0, 168.0]]
('大汉', 0.999447226524353)
[[52.0, 212.0], [148.0, 212.0], [148.0, 245.0], [52.0, 245.0]]
('hello world', 0.991887092590332)

**Comment by sulv9999** [comment]
今晚真滴很难过，我破防了

**Comment by GreatV** [comment]
看着像显存不够，换cpu跑吧

**Comment by sulv9999** [comment]
我的显存22g

> 看着像显存不够，换cpu跑吧

**Comment by GreatV** [comment]
@sulv9999  不用gpu可以跑吗

**Comment by GreatV** [comment]
https://www.paddlepaddle.org.cn/install/quick paddle安装的版本和这里是对应的吗

**Comment by sulv9999** [comment]
> @sulv9999 不用gpu可以跑吗

可以跑，但是我用gpu有时候成功，有时候失败，无论分配多少显存都是一样的

**Comment by sulv9999** [comment]
> https://www.paddlepaddle.org.cn/install/quick paddle安装的版本和这里是对应的吗

paddleocr版本和paddle版本需要对应吗，还是paddleocr包含paddleocr，我的paddleocr版本是2.7.3

**Comment by GreatV** [comment]
这个问题看起来和paddleocr无关，检查一下显存是否被占用，paddle和cuda版本是否对应。

**Comment by UserWangZz** [comment]
参考一下这个博客看看，是不是因为paddle没有安装好或者python版本的问题
https://blog.csdn.net/weixin_42921328/article/details/128543849
